Core/common_functions.py

```diff
@@ -0,0 +1,34 @@
+import os
+import subprocess
+import argparse
+from Core.Logger import logger
+
+def call_subprocess(cmd, args, env_vars={}):
+    child = subprocess.Popen(cmd.split(), env=env_vars, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    cmd_result = {"stdout":[],"stderr":[]}
+    while child.poll() is None:
+        for line in iter(child.stdout.readline, b''):
+            formated_string=line.decode().replace(line[:-1] if line[-1] == os.linesep else '\n', '')
+            cmd_result["stdout"].append(formated_string)
```

Comment:
If im not mistaken there is no concatenation here as i see it. It is rather formatting than concatenation. The last character is line separator and I just cut it before populating result into array. If you see how it might be improved please provide your suggestions.
As it goes about storing in memory results of the command execution with huge output - yes - that is not a best idea. But there were no requirements about that. we can have a loop containing subprocess.communicate() redirecting some portions of output to desired location. Than if needed we will work with the stored output. But it might be considered to be superfluous for this task. But I remain at your disposal about this. Just outline your vision as requirement and I'll attempt to implement.